src/app/component/eco-news/components/create-news/create-news.component.ts

```diff
@@ -91,9 +91,9 @@ export class CreateNewsComponent implements OnInit {
 
   public onSourceChange(): void {
     this.createNewsForm.get('source').valueChanges.subscribe(source => {
-      source.startsWith('http://') ||
-      source.startsWith('https://') ||
-      source.length === 0 ? this.isLink = false : this.isLink = true;
+      this.isLink = source.startsWith('http://') ||
+                    source.startsWith('https://') ||
+                    source.length === 0 ? false : true;
```

Comment:
Why do you need to check the length?
Can you describe the whole logic of `source.startsWith`?

Reply:
In fact, something strange with this code - any not empty string will be detected as a link. E.g., let we have
```js
source = 'bla-bla'
```
is
```js
source.startsWith('http://')
```
true? No. So, the first statement is `false`.
```js
source.startsWith('https://')
```
is `false` too. Next we have
```js
source.length === 0
```
is `false` and in this way
```js
source.length === 0 ? false : true
```
returns `true` and in general `false || false || true` returns `true` for any not empty string.

Reply:
<button type="submit" [disabled]="!createNewsForm.valid ||
                                                  isArrayEmpty ||
                                                  isLink ||
                                                  createEcoNewsService.isImageValid">Publish</button>
The variable 'isLink' is used in a template in condition in attribute 'disabled'.
So this button would be disabled if:
1) the form were not valid. or
2) the Array of chosen tags were empty. or
3) var 'isLink' were true
So, I think it variable should have another name: isNotLink.

Actualy I dont know why author check the length. It is not my code.
The startsWith() method determines whether a string begins with characters from another string, returning true or false, respectively.

Reply:
Ok, let's say the name of variable is `isNotLink`. It doesn't explain anything - you whatever have `true` for any not empty string. Actually it doesn't matter that it's not your code - now it's your task), so - investigate, figure it out and write well.

Reply:
this.isLink = source.startsWith('http://') ||
                    source.startsWith('https://') ||
                    source.length === 0 ? false : true;
if source = 'http://' this code returns false;